server/game/cards/locations/05/oldwyk.js

```diff
@@ -7,30 +7,42 @@ class OldWyk extends DrawCard {
         this.reaction({
             when: {
                 onChallenge: (event, challenge) =>
-                    challenge.attackingPlayer === this.controller
-                    && challenge.challengeType === 'power'
+                    challenge.attackingPlayer === this.controller &&
+                    challenge.challengeType === 'power' &&
+                    this.anyDrownedGodInDeadPile()
             },
             cost: ability.costs.kneelSelf(),
             handler: () => {
-                var card = _.last(this.controller.deadPile.filter(c => c.hasTrait('Drowned God')));
-
-                if(!card) {
-                    this.game.addMessage('{0} finds no character in their dead pile to put into play with {1}', this.controller, this);
-                    return;
-                }
+                let card = _.last(this.controller.deadPile.filter(c => c.hasTrait('Drowned God')));
 
                 this.controller.putIntoPlay(card);
                 this.game.currentChallenge.addAttacker(card);
 
-                this.game.addMessage('{0} uses {1} to put into play {2} as an attacker from their dead pile', this.controller, this, card);
+                this.game.addMessage('{0} kneels {1} to put {2} into play from their dead pile as an attacker', 
+                                      this.controller, this, card);
 
-                this.untilEndOfChallenge(ability => ({
-                    match: card,
-                    effect: ability.effects.returnToHandOrDeckBottom()
-                }));
+                this.game.once('afterChallenge:interrupt', (event, challenge) => this.resolveAfterChallenge(challenge, card));
```

Comment:
(I'm getting the sense that the ThronesDB FAQ may just be wrong here, since it definitely fits the definition of a Delayed Effect, which should happen at step 5 of the triggered condition resolution, just before Reactions but after Interrupts)